platform/mellanox/mlnx-fw-upgrade.j2

```diff
@@ -93,51 +192,148 @@ function UpgradeFW() {
     fi
 
     if [[ -z "${_FW_CURRENT}" ]]; then
-        echo "Could not retreive current FW version"
-        exit "${EXIT_ERROR}"
+        ExitFailure "could not retrieve current ASIC firmware version"
     fi
 
     if [[ -z "${_FW_AVAILABLE}" ]]; then
-        echo "Could not retreive available FW version"
-        exit "${EXIT_ERROR}"
+        ExitFailure "could not retrieve available ASIC firmware version"
+    fi
+
+    if [[ "${_FW_CURRENT}" = "${_FW_AVAILABLE}" ]]; then
+        LogInfo "ASIC firmware is up to date"
+        return "${EXIT_SUCCESS}"
+    fi
+
+    LogNotice "ASIC firmware upgrade is required. Installing compatible version..."
+
+    LogInfo "current ASIC firmware version: ${_FW_CURRENT}"
+    LogInfo "target ASIC firmware version: ${_FW_AVAILABLE}"
+
+    LogInfo "ASIC firmware file: ${_FW_FILE}"
+
+    RunCmd "${BURN_CMD} -i ${_FW_FILE}"
+}
+
+function UpgradeCPLDFW_Worker() {
+    local -r _CPLD_BURN_FILE="${1}"
+    local -r _CPLD_REFRESH_FILE="${2}"
+    local -r _ASIC_DEV="$(find /dev/mst -iname '*_pciconf0')"
+
+    if [[ "${UPDATE_MLNX_CPLD_FW}" = "1" ]]; then
```

Comment:
UPDATE_MLNX_CPLD_FW is not passed for sudo reboot during firmware update. can we have a file to indicate the cpld firmware is updated and refer to the file for the refresh command execution?

Reply:
@sujinmkang answered via the mail